src/sage/schemes/elliptic_curves/gp_simon.py

```diff
@@ -16,31 +16,19 @@
 #
 #                  https://www.gnu.org/licenses/
 # ****************************************************************************
+from pathlib import Path
 
-from sage.structure.parent_gens import localvars
+from cypari2.handle_error import PariError
 
-from sage.interfaces.gp import Gp
-from sage.misc.sage_eval import sage_eval
+from sage.env import SAGE_EXTCODE
+from sage.libs.pari import pari
 from sage.misc.randstate import current_randstate
 from sage.rings.integer_ring import ZZ
 from sage.rings.rational_field import QQ
+from sage.structure.parent_gens import localvars
 
 
-gp = None
-def init():
-    """
-    Function to initialize the gp process
-    """
-    global gp
-    if gp is None:
-        import os
-        from sage.env import DOT_SAGE
-        logfile = os.path.join(DOT_SAGE, 'gp-simon.log')
-        gp = Gp(script_subdirectory='simon', logfile=logfile)
-        gp.read("ellQ.gp")
-        gp.read("ell.gp")
-        gp.read("qfsolve.gp")
-        gp.read("resultant3.gp")
+simon_dir = Path(SAGE_EXTCODE) / 'pari' / 'simon'
```

Comment:
Wouldn't this be better in the function (in case something changes during running)? It's not really a big thing though, just a thought that crossed my mind

Reply:
sorry, I do not understand your concern. The `simon_dir` path is a clear constant.

Reply:
Ah, right, `SAGE_EXTCODE` should never change while Sage is running. Nevermind.